api/afdb/userName.ts

```diff
@@ -0,0 +1,23 @@
+"use server"
+
+import axios from 'axios';
+import getAxiosConfig from '../axiosConfig';
+
+export async function getUserName(studentId: string): Promise<string> {
+    const url = process.env.AF_DB_SERVICE_URL;
+    const bearerToken = process.env.AF_DB_SERVICE_BEARER_TOKEN || '';
```

Comment:
If token is empty, we shouldn't be in this method at all, so throw an error

Reply:
That condition is already adjusted in AuthContext file. This function won't get called if `userId` is invalid

Reply:
Then remove the condition `||` :)

Reply:
If I'll remove the || '', TypeScript raises an error because it cannot guarantee that bearerToken won't be undefined at runtime. This condition informs TypeScript that we are aware of the potential undefined value and have handled it explicitly.

Reply:
Sorry I don't understand. Then why is it not there for the previous line `const url=`?

Reply:
For `const url`, typeScript is able to infer its type correctly, so no need to provide a default value (as typescript gusses the type by itself). But sometimes there is a scenario where we have to explicity provide the defalut value (datatype) when it is not able to infer the type correctly. This might be the condition due to the complexity in the environment variables etc. You can read about it more here:
https://www.typescripttutorial.net/typescript-tutorial/typescript-type-inference/